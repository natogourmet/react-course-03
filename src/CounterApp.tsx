import PropTypes from 'prop-types';
import { useState } from 'react';

interface Props {
  value: number;
}

export const CounterApp = ({ value }: Props) => {
  const [count, setCount] = useState(value);

  const handleIncrement = (e: any) => {
    setCount((c) => c + 1);
  };

  const handleDecrement = (e: any) => {
    setCount((c) => c - 1);
  };

  const handleReset = (e: any) => {
    setCount(value);
  };

  return (
    <>
      <h1>Counter App</h1>
      <h2>{count}</h2>
      <button onClick={handleDecrement}>-1</button>
      <button aria-label="reset-btn" onClick={handleReset}>Reset</button>
      <button onClick={handleIncrement}>+1</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
