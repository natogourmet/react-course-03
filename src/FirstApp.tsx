import React from 'react';
import PropTypes from 'prop-types';

interface props {
  title: string;
  subtitle: string;
}

const name: string = 'Nato Gourmet';

export const FirstApp = ({ title, subtitle }: props) => {
  return (
    <>
      <h1>{title}</h1>
      <p>{subtitle}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  title: 'Missing Nato',
  subtitle: 'Not a Software Engineer',
}