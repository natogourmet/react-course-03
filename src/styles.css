html,
body {
  background-color: #282c34;
  color: white;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 1.3rem;
  padding: 70px;
}

button {
  background-color: #61afef;
  font-size: 1.3rem;
  padding: 15px;
  border: none;
  border-radius: 5px;
  color: white;
  cursor: pointer;
  margin: 1rem;
}
