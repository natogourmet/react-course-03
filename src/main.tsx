import React from 'react';
import { createRoot } from 'react-dom/client';
// import { App } from './HelloWorldApp';
import { FirstApp } from './FirstApp';
import './styles.css';
import { CounterApp } from './CounterApp';

createRoot(document.getElementById('root')!).render(
  <React.StrictMode>
    {/* <FirstApp title="Nato Gourmet" subtitle="Software Engineer" /> */}
    <CounterApp value={44} />
  </React.StrictMode>
);
