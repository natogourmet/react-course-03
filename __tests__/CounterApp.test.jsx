import { render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe('Tests on CounterApp', () => {
  
  const initValue = 100;
  
  test('should match with snapshot', () => {
    const { container } = render( <CounterApp value={initValue} /> );
    expect(container).toMatchSnapshot();
  });

  test('should contain initialValue', () => {
    render( <CounterApp value={initValue} /> );
    expect(screen.getByText(initValue.toString())).toBeTruthy();
  });
  
});