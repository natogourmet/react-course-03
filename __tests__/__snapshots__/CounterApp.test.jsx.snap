// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on CounterApp should match with snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    100
  </h2>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
  <button>
    +1
  </button>
</div>
`;
